File: perl/oauth_config.pl
# Perl Vanilla OAuth Client Example for UserSpice
# Configuration settings
# In a real application, this might be a .conf file or use a config module.

use strict;
use warnings;

# Subroutine to return configuration as a hash reference
sub get_oauth_settings {
    my %settings = (
        server_url        => "https://your-oauth-server.com/", # Example: "https://example.com/userspice/"
        client_id         => "your_client_id",
        client_secret     => "your_client_secret",
        redirect_uri      => "https://your_app_domain.com/oauth_response_perl", # Example: "http://localhost/callback.pl"

        auth_endpoint_path  => "usersc/plugins/oauth_server/auth.php",
        token_endpoint_path => "usersc/plugins/oauth_server/auth.php", # Often same as auth for UserSpice
    );
    return \%settings;
}

# This makes the file usable as a module or runnable to print config
if (!caller) {
    my $config = get_oauth_settings();
    print "OAuth Client Configuration (Perl):\n";
    foreach my $key (sort keys %$config) {
        print "$key: $config->{$key}\n";
    }
}

1; # Required for 'use' or 'require' to work

File: go/oauth_config.go
package main

// UserSpice OAuth Client Configuration
//
// Configure these settings to connect to your UserSpice OAuth server.
//
// To get your client credentials:
// 1. Log into your UserSpice admin panel
// 2. Navigate to OAuth Server settings
// 3. Create a new OAuth client application
// 4. Copy the client_id and client_secret
// 5. Set the redirect_uri to match exactly what you register

var oauthConfig = OAuthConfig{
	// The base URL of your UserSpice OAuth server (include trailing slash)
	ServerURL: "https://your-oauth-server.com/",

	// Your OAuth client ID (obtained from UserSpice admin panel)
	ClientID: "your_client_id",

	// Your OAuth client secret (keep this secure!)
	ClientSecret: "your_client_secret",

	// The URL users will be redirected to after authorization
	// This MUST match exactly what is registered in the OAuth server
	RedirectURI: "http://localhost:8080/oauth_response",

	// OAuth scopes to request (space-separated)
	// Available scopes: profile (includes user data)
	Scope: "profile",

	// HMAC Response Secret (optional but recommended for security)
	// If set, the OAuth server will sign the response data with HMAC-SHA256
	// and this client will verify the signature to prevent tampering.
	// This must match the 'response_secret' configured for your client in UserSpice.
	// Leave empty to disable signature verification.
	ResponseSecret: "",
}

// OAuthConfig holds the OAuth client configuration
type OAuthConfig struct {
	ServerURL      string
	ClientID       string
	ClientSecret   string
	RedirectURI    string
	Scope          string
	ResponseSecret string
}

// OAuth Endpoints (automatically derived from ServerURL):
//
// Authorization: {ServerURL}users/auth/
// Token:         {ServerURL}users/auth/
// UserInfo:      {ServerURL}users/auth/userinfo
